airflow/operators/python.py

```diff
@@ -728,33 +736,39 @@ def __init__(
             **kwargs,
         )
 
+    @property
+    def hook(self) -> PythonVirtualenvHook:
+        """Hook to manage the virtual environment."""
+        if self._hook is None:
+            self._hook = PythonVirtualenvHook(
+                venv_conn_id=self.venv_conn_id,
+                requirements=self.requirements,
+                python_version=self.python_version,
+                system_site_packages=self.system_site_packages,
+                pip_install_options=self.pip_install_options,
+                index_urls=self.index_urls,
+                venv_cache_path=self.venv_cache_path,
+                serializer=self.serializer,
+            )
+        return self._hook
+
     def _requirements_list(self, exclude_cloudpickle: bool = False) -> list[str]:
         """Prepare a list of requirements that need to be installed for the virtual environment."""
-        requirements = [str(dependency) for dependency in self.requirements]
-        if not self.system_site_packages:
-            if (
-                self.serializer == "cloudpickle"
-                and not exclude_cloudpickle
-                and "cloudpickle" not in requirements
-            ):
-                requirements.append("cloudpickle")
-            elif self.serializer == "dill" and "dill" not in requirements:
-                requirements.append("dill")
-        requirements.sort()  # Ensure a hash is stable
-        return requirements
+        warnings.warn(
+            "PythonVirtualenvOperator._requirements_list is deprecated",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.hook._requirements_list(exclude_cloudpickle=exclude_cloudpickle)
```

Comment:
This is a private method, in my view no need to deprecate. Whomever is using this outside of this code has bac luck.